mkat_fpga_tests/__init__.py

```diff
@@ -124,18 +124,8 @@ def halt_array(self):
     @property
     def katcp_rct(self):
         if self._katcp_rct is None:
-            try:
-                self.katcp_array_port = int(
+            self.katcp_array_port = int(
```

Comment:
You still need to catch the index error in the case where the array is not initialised, and perhaps raise a RuntimeError or something else sensible to explain what happened.